README.md

```diff
@@ -1072,6 +1072,306 @@ We recommend that you can write your own `CMakeUserPresets.json` file. You can i
 }
 ```
 
+
+## Cheat sheet: autotools vs. cmake
+
+Assume that `$oomph_home_dir` is the `oomph-lib` home directory:
+<table>
+<tr><th>autotools</th><th>cmake</th></tr>
+
+<tr>
+<td colspan="2">
+<b>- Build and install the entire library using a helper script:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run the autogen script which prompts
+# for configure options
+./autogen.sh
+
+</pre>
+</td>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run the hierher script which prompts
+# for configure options
+./something_similar.sh
+</pre>
+</td>
+</tr>
+
+
+<tr>
+<td colspan="2">
+<b>- Alternative: Build and install the entire library step by step:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run manual config/build/install 
+# sequence
+./configure hierher check this
+make 
+make install
+</pre>
+</td>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run manual config/build/install
+# sequence
+cmake -G Ninja -B build
+cmake --build build
+cmake --install build
+</pre>
+</td>
+</tr>
+
+<tr>
+<td colspan="2">
+<b>- Run all self-tests:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run the tests
+make check
+</pre>
+</td>
+<td>
+<pre>
+# Go to demo_drivers directory hierher Puneet: does this also do the tests in self_tests?
+cd $oomph_home_dir/demo_drivers<br>
```

Comment:
OK; does this suggest that we should provide a top-level test procedure (again a very thin wrapper around the cmake machinery) to run ctest in both? (to mimic "cd $oomph-home; make check")

Reply:
Hmm, it's a little bit tricky because of the current folder structure. We'd have to update the root `CMakeLists.txt` file to register these tests, but then we'd have to run these tests during the build stage of the main library (since we can't have one `CMakeLists.txt` to build and install the library and a separate one to just build/run the tests) which makes things messy. 

If we really want to combine the tests, we'd ideally put the `demo_drivers/` and `self_tests/` directories under a single subdirectory of the project. We could have a single `CMakeLists.txt` adjacent to these directories that would descend into both with a single build. The folder structure would look like so:

```bash
<root-oomph-dir>
- CMakeLists.txt
- src/
- example_and_tests/  # dummy name
  - CMakeLists.txt
  - demo_drivers/
  - self_tests/
- ...
```

Reply:
Note to self from discussion with @MatthiasHeilManchester:

**TODO:** move `self_tests/` to `demo_drivers/` directory.